### PR TITLE
chore(cli-integ): try to fix some integ test failures

### PR DESCRIPTION
**Pull request by corymhall**
While investigating cli test failures I found some things that need to be cleaned up.

1. Ensure that the bootstrap tests do not also run `ensureBootstrapped` since the test itself is responsible for bootstrapping.
2. Ensure that the correct context is passed to each test

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by corymhall** [comment]
Running through test pipeline...

**Comment by rix0rrr** [comment]
> Ensure that the bootstrap tests do not also run ensureBootstrapped since the test itself is responsible for bootstrapping.

This shouldn't matter, as `ensureBootstrapped()` should bootstrap the global CDKToolkit stack, while the bootstrapping tests themselves bootstrap a test-specific bootstrap stack.

Also doesn't hurt, but something is wrong if this is a load-bearing change.

File: packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/bootstrapping.integtest.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ integTest('can bootstrap without execution', withDefaultFixture(async (fixture)
   });
 
   expect(resp.Stacks?.[0].StackStatus).toEqual('REVIEW_IN_PROGRESS');
-}));
+}, false));
```
Comment (rix0rrr) [inline packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/bootstrapping.integtest.ts:20]: This is kind of ugly. Can we not add something like:

```ts
export interface DisableBootstrapContext {
  readonly bootstrapDisabled?: boolean;
}

export function withoutBootstrap<A extends object>(block: (context: A & DisableBootstrapContext) => void) {
  return () => block({ ...context, bootstrapDisabled: true });
}
```

And have `withCdkApp` check the `bootstrapDisabled` flag?

Comment (corymhall) [reply packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/bootstrapping.integtest.ts]: updated!

**Comment by corymhall** [comment]
> > Ensure that the bootstrap tests do not also run ensureBootstrapped since the test itself is responsible for bootstrapping.
> 
> This shouldn't matter, as `ensureBootstrapped()` should bootstrap the global CDKToolkit stack, while the bootstrapping tests themselves bootstrap a test-specific bootstrap stack.
> 
> Also doesn't hurt, but something is wrong if this is a load-bearing change.

Yeah I don't think this actually fixes any test failures. As I was investigating failures I found this stuff and figured I would clean it up while I was at it.

File: packages/@aws-cdk-testing/cli-integ/lib/with-cdk-app.ts [hunk 2]
```diff
@@ -135,6 +137,27 @@ export function withDefaultFixture(block: (context: TestFixture) => Promise<void
...
+   */
+  readonly disableBootstrap?: boolean;
+}
+
+/**
+ * To be used in place of `withDefaultFixture` when the test
+ * should not create the default bootstrap stack
+ */
+export function withoutBootstrap(block: (context: TestFixture & DisableBootstrapContext) => Promise<void>) {
+  return withAws<TestContext>(withCdkApp(block));
```
Comment (rix0rrr) [inline packages/@aws-cdk-testing/cli-integ/lib/with-cdk-app.ts:158]: You forgot to add `disableBootstrap`:

```suggestion
  return withAws<TestContext>(withCdkApp((context) => block({ ...context, disableBootstrap: true })));
```

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk-testing/cli-integ/lib/with-cdk-app.ts [hunk 3]
```diff
@@ -61,7 +63,9 @@ export function withCdkApp<A extends TestContext & AwsContext>(block: (context:
         });
       }
 
-      await ensureBootstrapped(fixture);
+      if (context.disableBootstrap) {
```
Comment (rix0rrr) [inline packages/@aws-cdk-testing/cli-integ/lib/with-cdk-app.ts:66]:
```suggestion
      if (!context.disableBootstrap) {
```

**Review by rix0rrr — CHANGES_REQUESTED** [review]

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**